homeassistant/components/sensor/serial.py

```diff
@@ -0,0 +1,78 @@
+"""
+Support for reading data from a serial port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.serial/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_PORT
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pyserial-asyncio==0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Serial Sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORT): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Serial sensor platform."""
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_PORT)
```

Comment:
Not sure but other use `device` instead `port`?

Reply:
Seems that it depends on the platform. [Acer switch](https://home-assistant.io/components/switch.acer_projector/) uses `filename` or [Particulate matter](https://home-assistant.io/components/sensor.serial_pm/) `serial_device`. Not real standard.

Reply:
According to Wikipedia the name should be `serial_port` while the Arduino IDE says `port`.